docs/src/token.md

```diff
@@ -476,3 +476,151 @@ have a balance of zero to be closed.
 
 ### Non-Fungible tokens
 An NTF is simply a token type where only a single token has been minted.
+
+## Wallet Integration Guide
+This section describes how to integrate SPL Token support into an existing
+wallet supporting native SOL.  It assumes a model whereby the user has a single
+system account as their **main wallet address** that they send and receive SOL
+from.
+
+Although all SPL Token accounts do have their own address on-chain, there's no
+need to surface these additional addresses to the user.
+
+There are two programs that are used by the wallet:
+* SPL Token program - generic program that is used by all SPL Tokens
+* SPL Associated Token Account program - this program defines the convention and the
+  provides the mechanism for mapping the user's wallet address to the associated
+  token accounts they hold.
+
+### Associated Token Account
+The associated token account convention allows all tokens to have the same
+destination address, allowing the user share their main wallet address to
+receive any SPL token.
```

Comment:
I find this sentence a bit confusing, because all tokens don't really have the same `destination address`, and it's incumbent on the wallet to derive the actual destination address (which is what the very next paragraph is about).